### PR TITLE
Misc. Updates

### PR DESCRIPTION
**Pull request by blakeneyops**
Updates to support integration into other contexts as well as standardization. This would be targeting a 0.0.6 alpha release.

### CODE REVIEW

File: certs/main.tf [hunk 1]
```diff
@@ -148,9 +213,9 @@ module "cluster" {
   region  = "${var.region}"
 
   # LC parameters
-  ami              = "${var.ami}"
+  ami              = "${coalesce(lookup(var.ami_region_lookup, var.region), var.ami_custom)}"
```
Comment (yuhunglin) [inline certs/main.tf]: 👍  ah nice, reflecting the lookup changes from the other cert gen module.

File: certs/variables.tf [hunk 2]
```diff
@@ -74,8 +75,14 @@ variable "s3_bucket_prefix" {
   default     = ""
 }
 
-variable "cidr_whitelist" {
+variable "vpn_whitelist" {
```
Comment (yuhunglin) [inline certs/variables.tf]: oh, a split whitelist? what's the benefit?

File: generate-certs/main.tf [hunk 3]
```diff
@@ -37,8 +36,8 @@ resource "aws_iam_role_policy" "s3_gen_certs" {
         "s3:PutObject"
       ],
       "Resource": [
-        "arn:aws:s3:::${replace(var.s3_root_path,"/(\/)+$/","")}",
```
Comment (yuhunglin) [inline generate-certs/main.tf]: ah, this fun regex helped take care of cases where the root path might have trailing slashes.

File: generate-certs/main.tf [hunk 4]
```diff
@@ -47,17 +46,17 @@ resource "aws_iam_role_policy" "s3_gen_certs" {
         "s3:List*"
       ],
       "Resource": [
-        "arn:aws:s3:::${element(split("/", var.s3_root_path), 0)}"
```
Comment (yuhunglin) [inline generate-certs/main.tf]: root path was also used to try and anticipate cases where the certs wasn't going to be mounted at the base of an s3 bucket.

File: generate-certs/main.tf [hunk 5]
```diff
@@ -79,63 +78,80 @@ EOF
...
-    key_ou            = "${var.key_ou}"
-    cert_key_name     = "${var.cert_key_name}"
-    key_country       = "${var.key_country}"
-    key_province      = "${var.key_province}"
-    active_clients    = "${var.active_clients}"
-    revoked_clients   = "${var.revoked_clients}"
-    openvpn_host      = "${var.openvpn_host}"
-    force_cert_regen  = "${var.force_cert_regen}"
-    s3_push_dryrun    = "${var.s3_push_dryrun}"
+    active_clients   = "${var.active_clients}"
```
Comment (yuhunglin) [inline generate-certs/main.tf]: 👍  🔤

File: generate-certs/templates/user_data.tpl [hunk 6]
```diff
@@ -2,20 +2,20 @@
 manage_etc_hosts: True
 
 runcmd:
-  - echo "S3_REGION=\"ap-northeast-1\"" > /etc/default/openvpn-cert-generator
```
Comment (yuhunglin) [inline generate-certs/templates/user_data.tpl]: ah yes, thanks

File: certs/variables.tf (hunk 2 above)
Comment (blakeneyops) [inline certs/variables.tf]: Just additional granularity if desired, as the SSH should really be an admin backdoor only.

File: generate-certs/main.tf (hunk 3 above)
Comment (blakeneyops) [inline generate-certs/main.tf]: Strangest thing, I cannot get Terraform to allow this now. Getting the following:

```
Error loading Terraform: Error downloading modules: module openvpn_server: Error parsing .terraform/modules/bf21773e112cafd6226a9e0ab47e5ce3/main.tf: At 193:59: illegal char escape
```

File: certs/variables.tf (hunk 2 above)
Comment (yuhunglin) [inline certs/variables.tf]: 👌

**Comment by blakeneyops** [comment]
Fixed. Turned out the only place it was complaining about the regex was as a variable input to the template. I moved the regex into the template and that resolved it.

**Comment by yuhunglin** [comment]
👌

File: generate-certs/main.tf (hunk 3 above)
Comment (cristobal23) [inline generate-certs/main.tf]: same. @blakeneyops what did you do to fix it?

**Pull request merged**